bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -169,6 +185,47 @@ func (executor *bridgeExecutor) StoreBatchFromMultiversX(batch *bridgeCore.Trans
 	return nil
 }
 
+// addBatchSCMetadataMvx fetches the logs containing sc calls metadata for the current batch
+func (executor *bridgeExecutor) addBatchSCMetadataMvx(ctx context.Context, transfers *bridgeCore.TransferBatch) (*bridgeCore.TransferBatch, error) {
+	if transfers == nil {
+		return nil, ErrNilBatch
+	}
+
+	events, err := executor.multiversXClient.GetBatchSCMetadata(ctx, transfers)
+	if err != nil {
+		return nil, err
+	}
+
+	for i, t := range transfers.Deposits {
+		transfers.Deposits[i], err = executor.addMetadataToTransferMvx(t, events)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return transfers, nil
+}
+
+// addMetadataToTransferMvx fetches the logs containing sc calls metadata for the current batch
+func (executor *bridgeExecutor) addMetadataToTransferMvx(transfer *bridgeCore.DepositTransfer, events []*transaction.Events) (*bridgeCore.DepositTransfer, error) {
+	for _, event := range events {
+		depositNonce, err := multiversx.ParseUInt64FromByteSlice(event.Topics[1])
+		if err != nil {
+			return nil, fmt.Errorf("%w while parsing batch ID", err)
+		}
+
+		if depositNonce == transfer.Nonce {
+			processData(transfer, event.Topics[7])
```

Comment:
Question: this reuses the process data function for the eth->mvx flow. The Ethereum contract will need also the MissingProtocolMarker if the SC data provided is empty?
This function also encodes the length before applying the SC data bytes. Will this work on the etherum contracts?
If not, I strongly suggest that this existing function should be renamed to something like `processDataForMvx` and create a new one called `processDataForEth`